src/services/widgets/widgets.svc.coffee

```diff
@@ -76,20 +76,52 @@ angular
 
     @updateWidgetSettings = (widget, needContentReload=true) ->
       widget.isEditMode = false
-
       if _.isEmpty(widget.settings)
         $log.warn("ImpacWidgetsSvc: Tried to update widget: #{widget.id} with no settings", widget)
         return false
 
+      paramPicker = _.find(widget.settings, {'key': 'params-picker', 'reach': 'dashboard'})
+
+      if (paramPicker)
+        paramPicker.reach = null;
+        return @updateAllSameWidgets(widget, 'params-picker', paramPicker)
+
       widget.isLoading = true if needContentReload
       meta = _.reduce(_.map(widget.settings, (set) -> set.toMetadata() ), (result, setMeta) -> angular.merge(result, setMeta))
-      
-      return _self.update(widget, { metadata: meta }).then(
+
+      widget.metadata = meta;
+
+      _self.update(widget, { metadata: meta }).then(
         (updatedWidget) ->
           if needContentReload
             _self.show(updatedWidget).finally( -> updatedWidget.isLoading = false )
       )
 
+    @updateAllSameWidgets = (widget, settingName, settingsPrototype) ->
+      sameWidgets = _.filter ImpacDashboardsSvc.getCurrentDashboard().widgets, (wgt) -> wgt.name == widget.name
+
+      _.each(sameWidgets, (wgt) ->
+        setting = _.find wgt.settings, {'key': settingName}
+        setting.setOptions settingsPrototype.getOptions()
+        _self.updateWidgetSettings wgt, true
+      )
+
+      dashboard = ImpacDashboardsSvc.getCurrentDashboard()
+
+      defaultWidget = _self.getDefaultWidget widget.name
+
+      newMeta = _.cloneDeep widget.metadata
+
+      if (defaultWidget)
+        defaultWidget.metadata = newMeta
+      else
+        dashboard.status_selection.push({'name': widget.name, 'metadata': newMeta})
+
+      ImpacDashboardsSvc.update dashboard.id, {'status_selection': dashboard.status_selection}
+
+    @getDefaultWidget = (widgetName)->
+      return _.find(ImpacDashboardsSvc.getCurrentDashboard().status_selection, {'name': widgetName})
+
```

Comment:
The overall logic of applying the setting to the dashboard + to the dashboard's widgets has to be reworked as there is a simpler way to do it that would be closer to the current behaviour. Here is the logical cycle that should be followed:
  1. In WidgetsSvc.updateWidgetSettings() function:
    - add a parameter "ignoreReach", defaulted to false, that will trigger the behaviour below when relevant 
    - in the loop that is done on the widget's settings:
      if the setting contains the property "reach" and if `setting.reach == 'dashboard'` and `ignoreReach` is false
      then call the function `Dashboard.updateSetting(dashboard, setting)` (see just below)

  2. Define `DashboardsSvc.updateSetting(DHB, SET)` function:
    - call `DashboardsSvc.update(DHB.id, setting.toMetadata())` => will save the metadata to the dashboard's model
    - loop on DHB.widgets
        - loop on each widget's settings
          - if `setting.toMetadata()` has the same key as `SET.toMetadata()` (setting passed has an argument),
          then that means that the widget has a setting of the same kind that the one we are trying to push to all the dashboard's widgets.
          - => `angular.merge(widget.metadata, SET.toMetadata())`
          - => `WidgetsSvc.initWidgetSettings(widget)` (to re-initialize the settings based on the new metadata)
          - => `WidgetsSvc.updateWidgetSettings(widget, ignoreReach=true)` (to push the new metadata to the widget's model)

  3. In the concerned widget directive, `initContext()` function:
    - options should be defaulted to the `widget.metadata` if defined, then to the `dashboard.medata` if defined
    - => Allows to:
      - keep a specific config for a widget that hasn't ticked the box
      - apply the dashboard's config for the new widgets added to the dashboard